File: Increasing Version Number/IncreasingVersionNumber.cpp
long getMinimumOperations(vector<int> versionNumbers) {
    int n = versionNumbers.size();
    assert(n >= 1 and n <= 2e5);
    for (auto x: versionNumbers)   assert(x >= 1 and x <= 1e9);
        
    long ans = 0, lst = versionNumbers[0];
    for(int i = 1; i < n; i++) {
        if (lst < versionNumbers[i]) {
            lst = versionNumbers[i];
            continue;
        }
        
        long x = ((lst - versionNumbers[i]) / (i + 1)) + 1;
        ans += x, lst = versionNumbers[i] + (i + 1) * x;
    }
    return ans;
}

File: Increasing Version Number/Increasing_Version_Number.py
def getMinimumOperations(versionNumbers):
    n = len(versionNumbers)
    pre = 0
    out = 0

    for i in range(n):
        if versionNumbers[i] <= pre:
            add = (pre + 1 - versionNumbers[i] + i) // (i + 1)
            out += add
            pre = versionNumbers[i] + add * (i + 1)
        else:
            pre = versionNumbers[i]

    return out